lua/devcontainer-cli/devcontainer_utils.lua

```diff
@@ -27,64 +55,89 @@ local on_fail = function(exit_code)
   vim.cmd("silent! :checktime")
 end
 
-local on_success = function()
+local _on_success = function()
   vim.notify("Devcontainer process succeeded!", vim.log.levels.INFO)
 end
 
 -- on_exit callback function to delete the open buffer when devcontainer exits
 -- in a neovim terminal
 ---@param code (integer) the exit code
-local on_exit = function(_, code, _)
+local _on_exit = function(code)
   if code == 0 then
-    on_success()
+    _on_success()
     return
   end
 
-  on_fail(code)
+  _on_fail(code)
 end
 
---- execute command
----@param cmd (string) the command to execute in the devcontainer terminal
-function M.exec_command(cmd)
-  vim.fn.termopen(
-    cmd,
-    {
-      on_exit = on_exit,
-      on_stdout = function(_, _, _)
-        if win ~= -1 then
-          vim.api.nvim_win_call(
-            win,
-            function()
-              vim.cmd("normal! G")
-            end
-          )
-        else
-          vim.notify("Executed " .. cmd)
-        end
-      end,
-    }
-  )
-  if buffer ~= -1 then
-    vim.api.nvim_set_current_buf(buffer)
-  else
-    vim.notify("No buffer created, therefore no output from command will be visible", vim.log.levels.WARN)
+-- check if the value is in the given table
+local function tableContains(tbl, value)
+  for _, item in ipairs(tbl) do
+    if item == value then
+      return true
+    end
   end
+
+  return false
 end
 
 -- create a new window and execute the given command
 ---@param cmd (string) the command to execute in the devcontainer terminal
-local function spawn_and_execute(cmd)
-  prev_win = vim.api.nvim_get_current_win()
-  win, buffer = windows_utils.open_floating_window(on_detach)
-  M.exec_command(cmd)
+---@param direction (string|nil) the placement of the window to be created (float, horizontal, vertical)
+---@param close_on_exit (boolean|nil) whether to close terminal when process exits
+local function _spawn_and_execute(cmd, direction, close_on_exit)
+  direction = vim.F.if_nil(direction, "float")
+  if tableContains(directions, direction) == false then
+    vim.notify("Invalid direction: " .. direction, vim.log.levels.ERROR)
+    return
+  end
+
+  -- create the terminal
+  _terminal = Terminal:new {
+    cmd = cmd,
+    hidden = false,
+    display_name = "devcontainer-cli",
+    direction = vim.F.if_nil(direction, "float"),
```

Comment:
enhancement: Currently the size of the vertical window seems a bit small, maybe it would be good to add the ability to add a size or add something to make it a bit larger. Terminal windows can be resized via the normal methods, so it isn't the end of the world.